src/main/java/hudson/remoting/Util.java

```diff
@@ -108,6 +108,55 @@ static String indent(String s) {
     }
 
     /**
+     * Check if given URL is in the exclusion list defined by the no_proxy environment variable.
+     * On most *NIX system wildcards are not supported but if one top domain is added, all related subdomains will also
```

Comment:
You mean HTTP 301/302 redirection? The goal of this function is **only** to determine whether the slave needs to use a proxy or not to reach its master. The actual selection of proxy address, redirection/connection to the proxy will be handled by other parts: `Util.openURLConnection()` relies on `java.net.URL.openConnection()` to do so, and the Engine does it manually: https://github.com/jenkinsci/remoting/blob/remoting-2.59/src/main/java/hudson/remoting/Engine.java#L411

I tried to mimic as much as possible the behaviour of well known tools as curl or wget which supports the `no_proxy` environment variable.